src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/ResourceBuilder.java

```diff
@@ -24,6 +24,10 @@
 
     Boolean stop(SSCO ssco, Resource resource);
 
+    List<Resource> buildNames(P po, int index, List<Resource> resources);
+
+    CreateResourceRequest buildCreateRequest(P po, List<Resource> res, List<Resource> buildNames, int index) throws Exception;
```

Comment:
Please at least try to use expressive variable names, what is po, res and buildNames here?